src/commands/mod.rs

```diff
@@ -736,6 +774,372 @@ impl DaemonControl {
+                e.insert(tx.0);
+            }
+
+            tx.input.push(bitcoin::TxIn {
+                previous_output: coin.outpoint,
+                sequence: bitcoin::Sequence::ENABLE_RBF_NO_LOCKTIME,
+                // TODO: once we move to Taproot, anti-fee-sniping using nSequence
+                ..bitcoin::TxIn::default()
+            });
+
+            // Populate the PSBT input with the information needed by signers.
+            let coin_desc = self.derived_desc(coin);
+            let witness_script = Some(coin_desc.witness_script());
+            let witness_utxo = Some(bitcoin::TxOut {
+                value: coin.amount.to_sat(),
+                script_pubkey: coin_desc.script_pubkey(),
+            });
+            let non_witness_utxo = spent_txs.get(&coin.outpoint).cloned();
+            let bip32_derivation = coin_desc.bip32_derivations();
+            psbt_ins.push(PsbtIn {
+                witness_script,
+                witness_utxo,
+                bip32_derivation,
+                non_witness_utxo,
+                ..PsbtIn::default()
+            });
+        }
+
+        // Finally, create the PSBT with all inputs and outputs, sanity check it and return it.
+        let psbt = Psbt {
+            unsigned_tx: tx,
+            version: 0,
+            xpub: BTreeMap::new(),
+            proprietary: BTreeMap::new(),
+            unknown: BTreeMap::new(),
+            inputs: psbt_ins,
+            outputs: psbt_outs,
+        };
+        sanity_check_psbt(&self.config.main_descriptor, &psbt)?;
+        // TODO: maybe check for common standardness rules (max size, ..)?
+
+        Ok(CreateSpendResult { psbt })
+    }
+
+    /// Create PSBT to replace the given transaction.
+    pub fn rbf_psbt(
+        &self,
+        txid: &bitcoin::Txid,
+        feerate_vb: u64,
+        is_cancel: bool,
+    ) -> Result<CreateSpendResult, CommandError> {
+        // The transaction to be replaced must be one of ours and so we can assume it signals replaceability.
+        // The inputs in the replacement transaction will be either those coins used as inputs in the transaction
+        // to be replaced or "confirmed" coins. Therefore, we can assume the transaction to be replaced is
+        // the only one directly conflicting with the replacement transaction.
+        let mut db_conn = self.db.connection();
+
+        let prev_psbt = db_conn
+            .spend_tx(txid)
+            .ok_or(CommandError::UnknownSpend(*txid))?;
+        let prev_coins = {
+            let prev_outpoints: &Vec<bitcoin::OutPoint> = &prev_psbt
+                .unsigned_tx
+                .input
+                .iter()
+                .map(|txin| txin.previous_output)
+                .collect();
+            let prev_coins = db_conn.coins_by_outpoints(prev_outpoints);
+            assert_eq!(prev_outpoints.len(), prev_coins.len());
+            prev_coins
+        };
```

Comment:
Yep, I found it was easier this way to keep track of which variables are needed later on. But in case you'd prefer consistency with the rest of the code, I could also avoid doing that.

Reply:
Interesting. No, you can keep doing it.